ksqldb-streams/src/main/java/io/confluent/ksql/execution/streams/materialization/KsqlMaterialization.java

```diff
@@ -142,15 +142,20 @@ public MaterializedWindowedTable windowed() {
         return table.get(key, partition);
       }
 
-      final Iterator<Row> result = table.get(key, partition).getRowIterator();
-
-      return KsMaterializedQueryResult.rowIterator(
-          Streams.stream(result)
-              .map(row -> filterAndTransform(row.key(), getIntermediateRow(row), row.rowTime())
-                  .map(v -> row.withValue(v, schema())))
-              .filter(Optional::isPresent)
-              .map(Optional::get)
-              .iterator());
+      final KsMaterializedQueryResult<Row> result = table.get(key, partition);
+      final Iterator<Row> iterator = Streams.stream(result.getRowIterator())
+          .map(row -> filterAndTransform(row.key(), getIntermediateRow(row), row.rowTime())
+              .map(v -> row.withValue(v, schema())))
+          .filter(Optional::isPresent)
+          .map(Optional::get)
+          .iterator();
+
+      if (result.getPosition().isPresent()) {
+        return KsMaterializedQueryResult.rowIteratorWithPosition(
+            iterator, result.getPosition().get());
+      }
+
+      return KsMaterializedQueryResult.rowIterator(iterator);
```

Comment:
It's significantly clearer if we have both branches of the if/else branch:
```suggestion
      if (result.getPosition().isPresent()) {
        return KsMaterializedQueryResult.rowIteratorWithPosition(
            iterator, result.getPosition().get());
      } else {
        return KsMaterializedQueryResult.rowIterator(iterator);
      }
```
This feedback applies elsewhere in this PR.